### PR TITLE
--read-length CLI option + some code suggestions

### PR DESCRIPTION
**Pull request by GabrielChanzy**
Hello,
Thanks so much for making your project available, I've really benefited from it :)
I've added an additional `--read-length` option for my use case (getting stats for live RTSP streams). My edits have been heavily influenced by [this other forked repo](https://github.com/unleashlive/stream-bitrate-stats). They seemed to stop maintaining it so thought to fork your project & initiate a pull request. (The gist of it is the `--read_intervals` option of [`ffprobe`](https://www.ffmpeg.org/ffprobe-all.html#:~:text=%2Dread_intervals))

The other commits are small edits I made in the code, the last one being a bug fix for the new `--read-length` option. Hopefully the commit message is clear enough; let me know if you require clarifications.

Thanks for taking the time to go through this. This is my first time trying out forks & pull requests so please let me know if I could have done something better! I am pretty unfamiliar with FFmpeg & python as well, so let me know if I misunderstood something :)

### CODE REVIEW

**Comment by slhck** [comment]
Thanks for the suggestion. I can see how this is useful!

Some thoughts: It imposes some restrictions on the value of `read_intervals`. While you might be able to specify `10%+20,01:30%01:45` in `ffprobe`, here we just allow a single number. I guess this is fine for most applications though.

I will leave some more comments in the code.

File: ffmpeg_bitrate_stats/bitrate_stats.py [hunk 1]
```diff
@@ -191,6 +199,10 @@ def _calculate_frame_sizes(self) -> list[FrameEntry]:
             "error",
             "-select_streams",
             self.stream_type[0] + ":0",
+            # "-analyzeduration",
+            # str(self.chunk_size*1000000),
```
Comment (slhck) [inline ffmpeg_bitrate_stats/bitrate_stats.py:203]: Please remove this if not needed.

Comment (GabrielChanzy) [reply ffmpeg_bitrate_stats/bitrate_stats.py:203]: From my tests, it seems unnecessary; I copied it over from unleashlive's repo. I'll remove it.

As a side note, do you understand what the `-analyzeduration` option does? From what I read, it seems to specify the duration that ffprobe's probe attempts to read the input file's metadata/headers. Is that right?

File: ffmpeg_bitrate_stats/bitrate_stats.py [hunk 2]
```diff
@@ -191,6 +199,10 @@ def _calculate_frame_sizes(self) -> list[FrameEntry]:
...
+            "-read_intervals",
+            f"%+{self.read_length}" if self.read_length != None else "%",
```
[the remark this answers is not in the document]
Comment (GabrielChanzy) [reply ffmpeg_bitrate_stats/bitrate_stats.py:205]: > I guess this works fine if nothing is specified?

Yes, `-read_intervals %` would read the whole file from start to finish.

**Comment by GabrielChanzy** [comment]
I've tested the latest commit with a video I ripped off youtube (it's 6+ mins long). 
image

> Some thoughts: It imposes some restrictions on the value of read_intervals. While you might be able to specify 10%+20,01:30%01:45 in ffprobe, here we just allow a single number. I guess this is fine for most applications though.

I've changed the `--read-length` option to `--read-start` & `--read-end`. Only takes in seconds for now. For read end, it will be relative duration rather than absolute (eg `-rs 1 -re 2` means start from 1 second onwards and read for 2 seconds).

Here is how the help looks like
image

**Comment by slhck** [comment]
That makes sense, running the tests again.

**Comment by slhck** [comment]
Actually, looking at it again, I will change the parameter to be `--read-duration` rather than `--read-end`, because it's not the end time that is specified. Otherwise it's fine to merge, will do so later, thanks for the PR!

**Comment by slhck** [comment]
I had to fix some types using `mypy`. Now I think it has no issues anymore, I pushed a new commit to this PR. Can you please check though if the functionality works as intended? When I specify:

```
python3 -m ffmpeg_bitrate_stats test/test.mp4 -rs 0.5 -v
```

I would expect the output to contain 12 frames only (the full 1 second file has 25 frames), but I still get 25?

**Comment by slhck** [comment]
Did you get a chance to check the functionality? I'd like to merge it but I'm not sure if this does what it is supposed to do.

**Comment by GabrielChanzy** [comment]
Hi sorry, I didn't have the time to come back to this PR this week. I should be able to get back to you next Tue, let you know then :)

**Comment by slhck** [comment]
Ah yes, it says in the docs:

> Note that seeking is not accurate, thus the actual interval start point may be different from the specified position. Also, when an interval duration is specified, the absolute end time will be computed by adding the duration to the interval start point found by seeking the file, rather than to the specified start value.

**Comment by slhck** [comment]
Released in v1.1.0, thanks for your contribution!

**Comment by slhck** [comment]
@all-contributors Please add @GabrielChanzy for code

**Pull request merged**